### PR TITLE
fix(dato-select): add translate & multi select disable

### PR DESCRIPTION
**Pull request by ritox842**

### CODE REVIEW

File: lib/src/options/option-template.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export function getOptionTemplate(isMulti = false) {
          dato-select__option--hover"
          [class.force-hide]="hide"
          ${isMulti ? '' : '[class.dato-option--active]="active"'}
-         [class.dato-select__option--disabled]="_disabled"
+         [datoTooltip]="getOptionTooltip()"
```
Comment (NetanelBasal) [inline lib/src/options/option-template.ts]: don't create a tooltip instance if it's disabled. Check if we have input for that or create one

File: lib/src/options/option.component.ts [hunk 2]
```diff
@@ -127,4 +123,10 @@ export class DatoOptionComponent implements OnInit {
       this.detectChanges();
     }
   }
+
+  getOptionTooltip() {
+    if (this.disabled) {
+      return this.active ? 'This item cannot be removed' : 'This item cannot be selected';
```
Comment (NetanelBasal) [inline lib/src/options/option.component.ts]: translate

File: lib/src/select/select.component.ts [hunk 3]
```diff
@@ -842,12 +846,17 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
       });
   }
 
-  ngOnDestroy() {
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: return it

File: lib/src/select/select.component.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ const valueAccessor = {
   changeDetection: ChangeDetectionStrategy.OnPush,
   exportAs: 'datoSelect'
 })
-export class DatoSelectComponent extends BaseCustomControl implements OnInit, OnDestroy, ControlValueAccessor, AfterContentInit {
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: return OnDestroy

File: lib/src/select/select.component.ts [hunk 5]
```diff
@@ -289,6 +289,9 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
   /** Whether is selectAll checked */
   _checked;
 
+  /*Holds all disabled options ID's**/
+  disabledIDs: string[] | number[];
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: (string | number)[]

File: lib/src/select/select.component.ts [hunk 6]
```diff
@@ -842,12 +846,17 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
...
-    this.infiniteSubscription && this.infiniteSubscription.unsubscribe();
-    this.keyboardEventsManagerSubscription && this.keyboardEventsManagerSubscription.unsubscribe();
-    this.datoOverlay.attached && this.datoOverlay.destroy();
+  /**
+   * Run through all datoOptionComponent and gather
+   * all disabled option ID's
+   */
+  private setDisabledIDs() {
+    this.disabledIDs = [];
+    this.options._results.forEach(optionComponent => {
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: for of

File: lib/src/select/trigger-multi/trigger-multi.component.html [hunk 7]
```diff
@@ -1,22 +1,24 @@
...
-       (click)="removeFocus($event)">
+    <div class="dato-trigger-multi__active d-flex justify-between"
+         *ngFor="let option of options | slice: 0: limitTo;"
+         (click)="removeFocus($event)">
     <span [ngStyle]="{maxWidth: maxActiveElementWidth}" [datoTooltip]="option[labelKey]"
           datoTooltipPosition="bottom" [datoTooltipOnTextOverflow]="true">{{option[labelKey]}}</span>
-    <dato-icon class="d-flex-no-shrink" datoIcon="close" (click)="remove(option); removeFocus($event)"></dato-icon>
-  </div>
+        <dato-icon class="d-flex-no-shrink" datoIcon="close"
+                   [style.visibility]="isDisabled(option)?'hidden':'inherit'" (click)="remove(option); removeFocus($event)"></dato-icon>
```
Comment (NetanelBasal) [inline lib/src/select/trigger-multi/trigger-multi.component.html]: spacing

Comment (NetanelBasal) [inline lib/src/select/trigger-multi/trigger-multi.component.html]: Don't use binding, use JS.

File: lib/src/select/trigger-multi/trigger-multi.component.ts [hunk 8]
```diff
@@ -19,45 +20,65 @@ import { defaultOptionsDisplayLimit } from '../select.types';
...
 
-  @ViewChild('input') input: ElementRef;
+  @ViewChild('input')
+  input: ElementRef;
 
   _restCount = 0;
+  @Input()
+  control: FormControl;
+  @Input()
+  disabledIDs: string[] | number[] = [];
```
Comment (NetanelBasal) [inline lib/src/select/trigger-multi/trigger-multi.component.ts]: type

File: lib/src/select/trigger-multi/trigger-multi.component.ts [hunk 9]
```diff
@@ -19,45 +20,65 @@ import { defaultOptionsDisplayLimit } from '../select.types';
...
   @HostBinding('class.dato-trigger-multi--focused')
   get isFocused() {
     return this._isFocused;
   }
+
   set isFocused(value: boolean) {
     this._isFocused = value;
   }
 
-  @Input() control: FormControl;
```
Comment (NetanelBasal) [inline lib/src/select/trigger-multi/trigger-multi.component.ts]: check why it's deleted

File: lib/src/options/option-template.ts (hunk 1 above)
Comment (ritox842) [reply lib/src/options/option-template.ts]: We need to check the tooltip together and deiced what is the best way to do it

**Pull request closed without merging**